engine/src/system_factory.cpp

```diff
@@ -0,0 +1,743 @@
+
+    if(object.attributes.count("override"))
+    {
+        string value = object.attributes["override"];
+        string::size_type eqpos = value.find_first_of('=');
+        if (eqpos != string::npos)
+        {
+            string name = value.substr(0,eqpos);
+            string override_value = value.substr(eqpos+1);
+            paramOverrides[name] = override_value;
+        }
+    }
+
+
+    //xml->fade saves if it's autogenerated system
+    // TODO: refactor this much later. Code smells.
+    if (isDestination == true && xml->fade)
+        radius = ScaleJumpRadius( radius );
+
+    radius *= xml->scale;
+
+    // Create Planet
+    // A lot of the issues with the old code were due to the fact each XML element was parsed separately
+    // with no awareness of other elements before. Therefore, state needed to be saved during processing.
+    // Here we see this with the planet "owner" - I assume it is what the planet is orbiting.
+
+    // Top level objects orbit the center?
+    QVector orbit_center = QVector(0,0,0);
+    if(owner == nullptr) {
+        // For top level, we introduce this instead
+        orbit_center = xml->cursun.Cast()+xml->systemcentroid.Cast();
+    }
+
+    Vector computed_rotational_velocity = ComputeRotVel( rotational_velocity, R, S );
+    Planet *planet = UnitFactory::createPlanet(R, S, velocity,
+                                               computed_rotational_velocity,
+                                               position, gravity, radius,
+                                               filename, technique, unitname,
+                                               blend_source, blend_destination, destination,
+                                               orbit_center, owner, ourmat, curlights, faction,
+                                               fullname, insideout );
+
+    if(owner == nullptr)
+    {
+        // The sun or another body orbiting the center of the system
+        // Does it really orbit or is it stationary?
+        planet->SetPosAndCumPos(R+S+xml->cursun.Cast()+xml->systemcentroid.Cast());
+        planet->SetOwner(getTopLevelOwner());
+        xml->moons.push_back(planet); // We store top level in moons and AddUnit them in
+        // star_system_xml
+    } else {
+        // It's a planet or moon or some other satellite
+        owner->satellites.prepend(planet);
+        planet->SetOwner(owner);
+        owner->SetOwner(getTopLevelOwner());
+    }
+
+    planet->applyTechniqueOverrides(paramOverrides);
+    return planet;
+}
+
+void SystemFactory::processSpaceElevator(Object& object, Planet* owner)
+{
+    string myfile = getStringAttribute(object, "file", "elevator");
+    string varname = getStringAttribute(object, "varname");
+    float  varvalue = getFloatAttribute(object, "varvalue", 0.0f);
+
+    Unit *unit = static_cast<Unit*>(owner);
+
+    if (owner == nullptr || unit->isUnit() != PLANETPTR) return;
```

Comment:
NOTE: C/C++ does not guarantee that the order of things in a evaluation like that is guaranteed. It might process the second part of the equation before the first part.

Reply:
This is true in general and incorrect in this case. TBF, I am a newbie C++ programmer and was ready to believe you when I decided to check.  From https://en.cppreference.com/w/cpp/language/eval_order : "_6) Every value computation and side effect of the first (left) argument of the built-in logical AND operator && and the built-in logical OR operator || is sequenced before every value computation and side effect of the second (right) argument._"

Reply:
New as of C++11. Thanks!